src/core/radio/radio_platform.cpp

```diff
@@ -87,6 +88,14 @@ extern "C" void otPlatRadioTxDone(otInstance *aInstance, otRadioFrame *aFrame, o
     Instance     &instance = AsCoreType(aInstance);
     Mac::TxFrame &txFrame  = *static_cast<Mac::TxFrame *>(aFrame);
     Mac::RxFrame *ackFrame = static_cast<Mac::RxFrame *>(aAckFrame);
+#if OPENTHREAD_RADIO
+    uint8_t channel = txFrame.mInfo.mTxInfo.mRxChannelAfterTxDone;
+
+    if ((otPlatRadioGetSupportedChannelMask(aInstance) & (1UL << channel)) != 0)
```

Comment:
We may not need this check? 
- OT core should already validate this 
- `otPlatRadioReceive` may also check this in its implemenation.

Reply:
Sounds reasonable. I was afraid the value was not correctly set. Perhaps we should add an assert here.